### PR TITLE
Consume a new snippet demonstrating tuples as out parameters

### LINKED ISSUE #11874: Tuples as out variables
It can be a nice example to have tuples as out variables.
`

    Dictionary<int, (int min, int max)> ranges;

    ...

    if (ranges.TryGetValue(key, out (int min, int max)))
    // similar to dict.TryGetValue(key, out int value)
`
I could not make this work, and there are lots of classes applying this pattern. If it cannot work, it will be nice to indicate that it is not possible.

EDIT: 
I found out from [this StackOverflow answer](https://stackoverflow.com/questions/51050951/c7-out-variables-with-tuples) that one can use 
`
    ranges.TryGetValue(key, out (int, int) minMax)
`
or
`
    ranges.TryGetValue(key, out (int min, int max) minMax)
`
which may or may not be added as an example.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3bf15268-9d34-cf10-97b9-0b94022c3305
* Version Independent ID: 17728168-48fa-0f5f-0c19-31ae4ce6fdb9
* Content: [Tuple types - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/tuples#feedback)
* Content Source: [docs/csharp/tuples.md](https://github.com/dotnet/docs/blob/master/docs/csharp/tuples.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by davidlfox**
## Summary

Is there a standardization doc for linking to 1) snippets and 2) other anchors within a doc? I'm not 100% sure how to _test_ my snippets e.g.

- `[_unnamed_](#named-and-unnamed-tuples)`
- `**[Deconstruction](#Deconstruction)**`

Happy to discuss/revise this further.

#### Update:
I was able to confirm the linked anchors' behavior (though there were build _warnings_) simply by viewing tuple.md in the browser at Github.com--still not 100% convinced

Fixes #11874

### CODE REVIEW

**Review by mairaw — COMMENTED** [review]
Depends on PR https://github.com/dotnet/samples/pull/1621 to be merged first. 

Thanks @davidlfox. I'll leave this review for @BillWagner but wanted to address some of your questions and provide feedback.

We recommend the following extension for VS Code to work with the Markdown files from docs:
https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack

It'll offer you some syntax help when inserting links, anchors, etc. and then the build report produced will tell if your PR has any issues.

**Comment by davidlfox** [comment]
@mairaw Confirmed dependency on https://github.com/dotnet/samples/pull/1621

**Comment by davidlfox** [comment]
Plugin preview feature confirms link/anchor behavior, though snippet/sample behavior still kind of an unknown. What process integrates snippets into docs Markdown files?

File: docs/csharp/tuples.md [hunk 1]
```diff
@@ -426,6 +426,20 @@ if (("Althea", "Goodwin") == p)
 
 The `Deconstruct` method could convert the `Person` object `p` to a tuple containing two strings, but it is not applicable in the context of equality tests.
 
+## Tuples as out parameters
+
+Tuples can be used as out parameters _themselves_. Not to be confused with any ambiguity previously mentioned in the [Deconstruction](#deconstruction) section. In a method call, you need only describe the tuple's shape:
+
+[!code-csharp[TuplesAsOutParameters](../../samples/snippets/csharp/tuples/program.cs#01_TupleAsOutVariable "Tuples as out parameters")]
```
Comment (BillWagner) [inline docs/csharp/tuples.md]: By convention, we use the `~` on samples (I know other examples don't yet, we're updating links as we update articles):

```suggestion
[!code-csharp[TuplesAsOutParameters](~/samples/snippets/csharp/tuples/program.cs#01_TupleAsOutVariable "Tuples as out parameters")]
```

File: docs/csharp/tuples.md [hunk 2]
```diff
@@ -426,6 +426,20 @@ if (("Althea", "Goodwin") == p)
 
 The `Deconstruct` method could convert the `Person` object `p` to a tuple containing two strings, but it is not applicable in the context of equality tests.
 
+## Tuples as out parameters
+
+Tuples can be used as out parameters _themselves_. Not to be confused with any ambiguity previously mentioned in the [Deconstruction](#deconstruction) section. In a method call, you need only describe the tuple's shape:
```
Comment (BillWagner) [inline docs/csharp/tuples.md]: Markdig prefers the `*` to `_` (our linter will start complaining soon):

```suggestion
Tuples can be used as out parameters *themselves*. Not to be confused with any ambiguity previously mentioned in the [Deconstruction](#deconstruction) section. In a method call, you need only describe the tuple's shape:
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making this fix @davidlfox 

I had a couple small suggestions for style. Once you accept those suggestions, I'll :shipit:

Thanks for contributing to docs! 🎆

File: docs/csharp/tuples.md (hunk 1 above)
Comment (davidlfox) [reply docs/csharp/tuples.md]: Good to know--looking forward to helping docs an samples with more PR's!

**Comment by mairaw** [comment]
We have the samples repo configured as a dependent repo for this one. So once a PR is merged there, we can start seeing the changes here. @BillWagner this is ready to review.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @davidlfox This looks great, and I'm ready to :shipit:

**Pull request merged**